atd-vze/src/Components/GridTable.js

```diff
@@ -106,15 +192,44 @@ const GridTable = ({ title, query }) => {
    *
    **/
 
-  // Manage the WHERE clause of our query
-  query.cleanWhere(); // Clean slate
-  if (searchParameters["column"] && searchParameters["value"]) {
+  const isCrashesPage = query.table === "atd_txdot_crashes";
+
+  // Handle Date Range (only if available)
+  if (
+    isCrashesPage &&
+    dateRangeFilter["startDate"] &&
+    dateRangeFilter["endDate"]
+  ) {
     query.setWhere(
-      searchParameters["column"],
-      `_ilike: "%${searchParameters["value"]}%"`
+      "crash_date",
+      `_gte: \"${dateRangeFilter["startDate"]}\", _lte: \"${
+        dateRangeFilter["endDate"]
+      }\"`
     );
   }
 
+  // First initialize the filters
+  query.loadFilters(filters, filterOptions);
+
+  // Manage the WHERE clause of our query
+  if (searchParameters["column"] && searchParameters["value"]) {
+    // We will need to be careful which operator we will be using depending
+    // on whether the column is an integer or a string, etc.
+    if (searchParameters["column"] === "crash_id") {
```

Comment:
When I see nested if statements, its one signal that the logic is getting complex and it might be worth naming the conditions as a boolean variable. 

I would try something like:

```js
const hasColumnAndValue = searchParameters["column"] && searchParameters["value"]
const isColumnEqualToCrashId = searchParameters["column"] === "crash_id"

if (hasColumnAndValue && isColumnEqualToCrashId) {
...
} else if (hasColumnAndValue) {
...
}
```

I'm pretty sure that would be equivalent logic and might be easier to read without the nesting.